lib/reservation_booker.rb

```diff
@@ -0,0 +1,81 @@
+require_relative "reservation"
+require_relative "all_rooms"
+require_relative "date_range"
+require_relative "room"
+
+module Hotel
+  class ReservationBooker
+    attr_reader :all_reservations
+
+    def initialize
+      @all_rooms = Hotel::AllRooms.new
+      @all_reservations = []
+    end 
+
+    def create_new_reservation(check_in_year, check_in_month, check_in_day, check_out_year, check_out_month, check_out_day)
+      is_valid = Hotel::DateRange.valid_checkin_dates?(check_in_year, check_in_month, check_in_day)
+      if !is_valid 
+        raise ArgumentError, "Invalid Dates"
+      end 
+
+      is_valid = Hotel::DateRange.valid_checkout_dates?(check_out_year, check_out_month, check_out_day)
+      if !is_valid
+        raise ArgumentError, "Invalid Dates"
+      end 
+
+      check_in_date = Hotel::DateRange.check_in_date(check_in_year, check_in_month, check_in_day)
+      check_out_date = Hotel::DateRange.check_out_date(check_out_year, check_out_month, check_out_day)
+
+      is_valid = Hotel::DateRange.valid_dates?(check_in_date, check_out_date)
+      if !is_valid
+        raise ArgumentError, "Invalid Dates"
+      end 
+
+
+      rooms_not_reserved = available_rooms_by_date_range(check_in_year, check_in_month, check_in_day, check_out_year, check_out_month, check_out_day)
+      room = rooms_not_reserved.sample
+      
+      if room.available_on_these_dates?(check_in_date, check_out_date) == false
```

Comment:
This check should be taken care of if `available_rooms_by_date_range` can be trusted. But a good check to put here: what if there are no rooms left?